src/opencost_parquet_exporter.py

```diff
@@ -243,63 +268,53 @@ def save_result(processed_result, config):
     Returns:
     - uri : String with the path where the data was saved.
     """
-    file_name = 'k8s_opencost.parquet'
-    window = datetime.strptime(config['window_start'], "%Y-%m-%dT%H:%M:%SZ")
-    parquet_prefix = f"{config['file_key_prefix']}/year={window.year}"\
-                     f"/month={window.month}/day={window.day}"
-    try:
-        if config.get('s3_bucket', None):
-            uri = f"s3://{config['s3_bucket']}/{parquet_prefix}/{file_name}"
-        else:
-            uri = f"file://{parquet_prefix}/{file_name}"
-            path = '/'+parquet_prefix
-            os.makedirs(path, 0o750, exist_ok=True)
-        processed_result.to_parquet(uri)
-        return uri
-    except pd.errors.EmptyDataError as ede:
-        print(f"Error: No data to save, the DataFrame is empty.{ede}")
-    except KeyError as ke:
-        print(f"Missing configuration key: {ke}")
-    except ValueError as ve:
-        print(f"Error parsing date format: {ve}")
-    except FileNotFoundError as fnfe:
-        print(f"File or directory not found: {fnfe}")
-    except PermissionError as pe:
-        print(f"Permission error: {pe}")
-    except boto_exceptions.NoCredentialsError:
-        print("Error: No AWS credentials found to access S3")
-    except boto_exceptions.PartialCredentialsError:
-        print("Error: Incomplete AWS credentials provided for accessing S3")
-    except boto_exceptions.ClientError as ce:
-        print(f"AWS Client Error: {ce}")
-    return None
+    # TODO: Handle save to local file system. Make it default maybe?
+    storage = get_storage(storage_backend=config['storage_backend'])
+    uri = storage.save_data(data=processed_result, config=config)
+    if uri:
+        print(f"Data successfully saved at: {uri}")
+    else:
+        print("Failed to save data.")
+        sys.exit(1)
+
+# pylint: disable=C0116
+
 
 def main():
-    """
-    Main function to execute the workflow of fetching, processing, and saving data
-    for yesterday.
-    """
+    # TODO: Error handling when load fails
     print("Starting run")
+    print("Load data types")
+    data_types = load_config_file(
+        file_path='./src/data_types.json')  # TODO: Make path ENV var
```

Comment:
Please update the Dockerfile file to support this.  
The dockerfile need major update for this PR also because of the new files under src/ used for azure. 

Remember to also change the examples kubernetes resource if the entry point changes.

Reply:
Hi @lmello,
I've updated the PR. Was not 100% sure how you envisioned allowing for changes of `ENTRYPOINT` so please have look and let me know if you had something different in mind. I hope the changes address all requested changes.